### PR TITLE
Fix for Play/pause operation for DUT

### PR DESCRIPTION
**Pull request by nitinrawat67**
Play pause operation was not working from DUT as reference
device was not sending playbackstatus changed notification.
Added code to handle this at DUT side itself using
getplaystatus response.

Tracked-On: OAM-72645
Signed-off-by: Nitin Rawat <nitin.rawat@intel.com>

### CODE REVIEW

**Comment by nitinrawat67** [comment]
@JeevakaPrabu

File: android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
++static int play_flag=0;
++static int pause_flag=0;
+ /*****************************************************************************
+  *  Static functions
+  *****************************************************************************/
+@@ -3204,6 +3207,7 @@ static void handle_notification_response(tBTA_AV_META_MSG* pmeta_msg,
+          * if the play state is playing.
+          */
+         if (p_rsp->param.play_status == AVRC_PLAYSTATE_PLAYING ||
++	    p_rsp->param.play_status == AVRC_PLAYSTATE_PAUSED  ||
```
Comment (JeevakaPrabu) [inline android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch]: -Follow the same indentation as other lines
-Remove extra space between PAUSED and ||

File: android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+             p_rsp->param.play_status == AVRC_PLAYSTATE_REV_SEEK ||
+             p_rsp->param.play_status == AVRC_PLAYSTATE_FWD_SEEK) {
+           rc_start_play_status_timer(p_dev);
+@@ -3309,7 +3313,7 @@ static void handle_notification_response(tBTA_AV_META_MSG* pmeta_msg,
+         /* Start timer to get play status periodically
+          * if the play state is playing.
+          */
+-        if (p_rsp->param.play_status == AVRC_PLAYSTATE_PLAYING) {
++        if ((p_rsp->param.play_status == AVRC_PLAYSTATE_PLAYING)||(p_rsp->param.play_status == AVRC_PLAYSTATE_PAUSED)) {
```
Comment (JeevakaPrabu) [inline android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch]: - introduce a space between ) and ||
- introduce a space between || and (
- If number of characters/line is exceeding the conventions followed in the file, then split it into multiple lines

File: android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+           rc_start_play_status_timer(p_dev);
+           get_element_attribute_cmd(AVRC_MAX_NUM_MEDIA_ATTR_ID, attr_list,
+                                     p_dev);
+@@ -3869,6 +3873,24 @@ static void handle_get_playstatus_response(tBTA_AV_META_MSG* pmeta_msg,
+         FROM_HERE,
+         base::Bind(bt_rc_ctrl_callbacks->play_position_changed_cb,
+                    p_dev->rc_addr, p_rsp->song_len, p_rsp->song_pos));
++        if((p_rsp->song_pos ==0)&&(p_rsp->play_status ==0x02 )&& !pause_flag){
```
Comment (JeevakaPrabu) [inline android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch]: * introduce a space between if and (
* introduce a space between ) and &&
* introduce a space between && and (
* introduce a space between ) and {
* introduce a space between == and 0

File: android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
++           do_in_jni_thread(
++           FROM_HERE,
++           base::Bind(bt_rc_ctrl_callbacks->play_status_changed_cb,
++                       p_dev->rc_addr,
++                       (btrc_play_status_t)p_rsp->play_status));
++           pause_flag=1;
++           play_flag=0;
++
++        }
++       if((p_rsp->song_pos !=0)&&(p_rsp->play_status ==0x01) && !play_flag){
```
Comment (JeevakaPrabu) [inline android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch]: previous comment applies here too

File: android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                    p_dev->rc_addr, p_rsp->song_len, p_rsp->song_pos));
++        if((p_rsp->song_pos ==0)&&(p_rsp->play_status ==0x02 )&& !pause_flag){
++           do_in_jni_thread(
++           FROM_HERE,
++           base::Bind(bt_rc_ctrl_callbacks->play_status_changed_cb,
++                       p_dev->rc_addr,
++                       (btrc_play_status_t)p_rsp->play_status));
++           pause_flag=1;
++           play_flag=0;
++
```
Comment (JeevakaPrabu) [inline android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch]: remove extra blank line

File: android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+         FROM_HERE,
+         base::Bind(bt_rc_ctrl_callbacks->play_position_changed_cb,
+                    p_dev->rc_addr, p_rsp->song_len, p_rsp->song_pos));
++        if((p_rsp->song_pos ==0)&&(p_rsp->play_status ==0x02 )&& !pause_flag){
++           do_in_jni_thread(
++           FROM_HERE,
++           base::Bind(bt_rc_ctrl_callbacks->play_status_changed_cb,
++                       p_dev->rc_addr,
++                       (btrc_play_status_t)p_rsp->play_status));
++           pause_flag=1;
```
Comment (JeevakaPrabu) [inline android_p/google_diff/cel_apl/system/bt/0004-Fix-for-Play-pause-operation-for-DUT.patch]: - introduce space between pause_flag and =
introduce space between = and 1

**Pull request closed without merging**